public/components/experiment_view/hybrid_optimizer_experiment_view.tsx

```diff
@@ -87,22 +87,15 @@ export const HybridOptimizerExperimentView: React.FC<HybridOptimizerExperimentVi
             .then(sanitizeResponse));
 
         if (_experiment && _searchConfiguration && _querySet && _judgmentSet) {
-          // Get all evaluation result IDs
-          const evaluationResultIds = _experiment.results
-            .flatMap((res) => {
-              return res.evaluationResults.map(({evaluationId}) => evaluationId);
-            })
-            .filter(Boolean);
-
-          // Fetch all evaluation results in one query
+          const querySetSize = _querySet && Object.keys(_querySet.querySetQueries).length;
           const query = {
             index: 'search-relevance-evaluation-result',
             query: {
-              terms: {
-                _id: evaluationResultIds,
-              },
+              match: {
+                experimentId: _experiment.id,
+              }
             },
-            size: evaluationResultIds.length,
+            size: querySetSize * 66,
```

Comment:
this is potentially unsafe assumption, we probably increase number of variants in future, and even now if some search requests failed then it's not gonna be 66. Is there a way to make this dynamic?

Reply:
Good point, the number of experiment variants could be stored in the experiment document and retrieved from there, wdyt?

Reply:
We will extract the number of variants from the experiment document once it records the number of successful and failed variants. At the moment 66 is a hard coded number that will also work in case of failures as it is used as an upper bound when retrieving results.